File: SelectionSort.js
// Function to perform selection sort on an array
function selectionSort(arr) {
  // Get the length of the array
  const n = arr.length;

  // Loop over each element of the array
  for (let i = 0; i < n - 1; i++) {
    // Assume the current element is the minimum
    let minIndex = i;

    // Find the index of the minimum element in the unsorted part of the array
    for (let j = i + 1; j < n; j++) {
      // If the current element is less than the current minimum, update the minimum
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    // If the minimum element is not the current element, swap them
    if (minIndex !== i) {
      [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
    }
  }

  // Return the sorted array
  return arr;
}

// Define an unsorted array
const unsortedArray = [64, 25, 12, 22, 11];

// Sort the array using the selection sort algorithm
const sortedArray = selectionSort(unsortedArray);

// Log the sorted array to the console
console.log(sortedArray); // [11, 12, 22, 25, 64]

File: SelectionSort.py
# This function performs a selection sort on an array
def selection_sort(arr):
    # Get the length of the array
    n = len(arr)

    # Iterate over each element in the array
    for i in range(n):
        # Assume the current element is the smallest
        min_idx = i

        # Iterate over the rest of the array
        for j in range(i+1, n):
            # If a smaller element is found, update the minimum index
            if arr[j] < arr[min_idx]:
                min_idx = j

        # Swap the current element with the smallest element found
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

    # Return the sorted array
    return arr
